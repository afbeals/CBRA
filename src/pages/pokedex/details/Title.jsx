// Extenal
import React from 'react';
import PropTypes from 'prop-types';

// Constants
const classname = 'title';

// Component
const Title = ({ pClassname, name, id }) => (
  <div data-testid="details-title" className={`${pClassname}__${classname}`}>
    <h3 className={`${pClassname}__${classname}__name`}>{id && <>{name}</>}</h3>
    <h3 className={`${pClassname}__${classname}__id`}>
      {id && (
        <>
          No. <span>{id}</span>
        </>
      )}
    </h3>
  </div>
);

Title.propTypes = {
  pClassname: PropTypes.string.isRequired,
  name: PropTypes.string,
  id: PropTypes.number,
};

export default Title;

export { Title };
