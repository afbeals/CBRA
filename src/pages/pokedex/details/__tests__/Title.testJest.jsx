// External
import React from 'react';
import { render, cleanup } from '@testing-library/react';

// Local
import Title from '../Title';

// tests
describe('Title', () => {
  afterEach(cleanup);
  it('should match Title to snapshot', () => {
    const { getByTestId } = render(<Title pClassname={'parent'} />);
    const title = getByTestId('details-title');
    expect(title).toMatchSnapshot();
  });

  it('should find a title for the pokemon', () => {
    const { getByText } = render(
      <Title pClassname={'parent'} name="magikarp" id={23} />,
    );
    const title = getByText('magikarp');
    const id = getByText('23');
    expect(title).toBeTruthy();
    expect(id).toBeTruthy();
  });
});
