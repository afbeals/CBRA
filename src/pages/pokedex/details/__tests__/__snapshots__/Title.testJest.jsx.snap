// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match Title to snapshot 1`] = `
<div
  class="parent__title"
  data-testid="details-title"
>
  <h3
    class="parent__title__name"
  />
  <h3
    class="parent__title__id"
  />
</div>
`;
